hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestNNThroughputBenchmark.java

```diff
@@ -60,6 +60,18 @@ public void testNNThroughput() throws Exception {
     NNThroughputBenchmark.runBenchmark(conf, new String[] {"-op", "all"});
   }
 
+  @Test
+  public void testNNThroughputWithBaseDir() throws Exception {
+    Configuration conf = new HdfsConfiguration();
+    conf.setInt(DFSConfigKeys.DFS_BLOCK_SIZE_KEY, 16);
+    File nameDir = new File(MiniDFSCluster.getBaseDirectory(), "name");
+    conf.set(DFSConfigKeys.DFS_NAMENODE_NAME_DIR_KEY,
+        nameDir.getAbsolutePath());
+    DFSTestUtil.formatNameNode(conf);
+    NNThroughputBenchmark.runBenchmark(conf,
+        new String[] {"-op", "all", "-baseDirName", "/nnThroughputBenchmark1"});
```

Comment:
test should check ``nnThroughputBenchmark1 `` was used & the default directory wasn't created nor used.

try with other operations as well apart from ``all``